client/state/sites/actions.js

```diff
@@ -124,11 +124,15 @@ export function requestSite( siteFragment ) {
 			.site( siteFragment )
 			.get()
 			.then( site => {
-				dispatch( receiveSite( omit( site, '_headers' ) ) );
-				dispatch( {
-					type: SITE_REQUEST_SUCCESS,
-					siteId: siteFragment,
-				} );
+				// If we can't manage the site, don't add it to state.
+				if ( site && site.capabilities ) {
```

Comment:
When we can't manage a site, I believe we also need to dispatch a `SITE_REQUEST_FAILURE` since we use it for requesting state in the reducer.

Reply:
I was thinking about this but wasn't sure because in [your original code](https://github.com/Automattic/wp-calypso/pull/21894/files#diff-e14cc80ab123f46e868d30ca5e17581fR81), we don't send `SITE_REQUEST_FAILURE`, we just return the previous state.

Reply:
Fixed in f0dde9f. Can you re-check please @gwwar?